src/SBInclinedExponential.cpp

```diff
@@ -0,0 +1,367 @@
+            fillKValueQuadrant(val,kx0,dkx,izero,ky0,dky,jzero);
+        } else {
+            xdbg<<"Non-Quadrant\n";
+            assert(val.stepi() == 1);
+            const int m = val.colsize();
+            const int n = val.rowsize();
+            typedef tmv::VIt<std::complex<double>,1,tmv::NonConj> It;
+
+            kx0 *= _r0;
+            dkx *= _r0;
+            ky0 *= _r0_cosi;
+            dky *= _r0_cosi;
+
+            for (int j=0;j<n;++j,ky0+=dky) {
+                double kx = kx0;
+                It valit = val.col(j).begin();
+                for (int i=0;i<m;++i,kx+=dkx) {
+                    double new_val = _flux * _info->kValue(kx,ky0);
+                    xxdbg << "kx = " << kx << "\tky = " << ky0 << "\tval = " << new_val << std::endl;
+                    *valit++ = _flux * _info->kValue(kx,ky0);
+                }
+            }
+        }
+    }
+
+    void SBInclinedExponential::SBInclinedExponentialImpl::fillKValue(tmv::MatrixView<std::complex<double> > val,
+                                            double kx0, double dkx, double dkxy,
+                                            double ky0, double dky, double dkyx) const
+    {
+        dbg<<"SBInclinedExponential fillKValue\n";
+        dbg<<"kx = "<<kx0<<" + i * "<<dkx<<" + j * "<<dkxy<<std::endl;
+        dbg<<"ky = "<<ky0<<" + i * "<<dkyx<<" + j * "<<dky<<std::endl;
+        assert(val.stepi() == 1);
+        assert(val.canLinearize());
+        const int m = val.colsize();
+        const int n = val.rowsize();
+        typedef tmv::VIt<std::complex<double>,1,tmv::NonConj> It;
+
+        kx0 *= _r0;
+        dkx *= _r0;
+        dkxy *= _r0;
+        ky0 *= _r0_cosi;
+        dky *= _r0_cosi;
+        dkyx *= _r0_cosi;
+
+        It valit = val.linearView().begin();
+        for (int j=0;j<n;++j,kx0+=dkxy,ky0+=dky) {
+            double kx = kx0;
+            double ky = ky0;
+            for (int i=0;i<m;++i,kx+=dkx,ky+=dkyx) {
+                *valit++ = _flux * _info->kValue(kx,ky);
+            }
+        }
+    }
+
+    double SBInclinedExponential::SBInclinedExponentialImpl::maxK() const { return _info->maxK() * _inv_r0; }
+    double SBInclinedExponential::SBInclinedExponentialImpl::stepK() const { return _info->stepK() * _inv_r0; }
+
+    InclinedExponentialInfo::InclinedExponentialInfo(double h_tani_over_r, const GSParamsPtr& gsparams) :
+        _h_tani_over_r(h_tani_over_r),
+        _half_pi_h_tani_over_r(0.5*M_PI*h_tani_over_r),
+        _gsparams(gsparams),
+        _maxk(0.), _stepk(0.)
+    {
+        dbg<<"Start InclinedExponentialInfo constructor for h_tani_over_r = "<<h_tani_over_r<<std::endl;
+
+        assert(h_tani_over_r >= 0); // Should only ever have non-negative
+
+        // For large k, we clip the result of kValue to 0.
+        // We do this when the correct answer is less than kvalue_accuracy.
+        // (1+k^2 r0^2)^-1.5 = kvalue_accuracy
+        _ksq_max = (std::pow(gsparams->kvalue_accuracy,-1./1.5)-1.);
```

Comment:
Is this calculation still correct?  I suspect the inclined disks need to go to higher k than face-on disks.  More power at small scales, which is higher k values.  Is the `_ksq_max` value that we used for Exponential still the relevant value here?  I would assume not, but maybe it's close?  At least do a bit more investigation of where the k values drop below `kvalue_accuracy` for various choices of inclination angle to see if you can come up with a heuristic or approximate choice for where it's ok to cut it off to zero.